cmd/integration/commands/stages.go

```diff
@@ -843,6 +872,44 @@ func stageBodies(db kv.RwDB, ctx context.Context, logger log.Logger) error {
 	return nil
 }
 
+func stagePolygonSync(db kv.RwDB, ctx context.Context, logger log.Logger) error {
+	engine, _, stageSync, _, _ := newSync(ctx, db, nil /* miningConfig */, logger)
+	heimdallClient := engine.(*bor.Bor).HeimdallClient
+	sn, borSn, agg, _ := allSnapshots(ctx, db, logger)
+	defer sn.Close()
+	defer borSn.Close()
+	defer agg.Close()
+	br, bw := blocksIO(db, logger)
+	dirs := datadir.New(datadirCli)
+	chainConfig := fromdb.ChainConfig(db)
+
+	return db.Update(ctx, func(tx kv.RwTx) error {
+		if reset {
+			if err := reset2.ResetPolygonSync(tx, db, agg, br, bw, dirs, *chainConfig, logger); err != nil {
+				return err
+			}
+			return nil
+		}
+
+		stageState := stage(stageSync, tx, nil, stages.Senders)
+		cfg := stagedsync.NewPolygonSyncStageCfg(logger, chainConfig, nil, heimdallClient, nil, 0, nil, br, nil, 0) // we only need blockReader and blockWriter (blockWriter is constructed in NewPolygonSyncStageCfg)
```

Comment:
If we need blockWriter why are we not passing it to the function?

Reply:
I'm not sure I understand the comment

Reply:
`blockWriter` is used in `ResetPolygonSync`